frontend/src/features/prototype/hooks/usePartDragSystem.ts

```
@@ -0,0 +1,255 @@
+      const newOriginalPositions = newSelected.reduce(
+        (acc, id) => {
+          const part = parts.find((pt) => pt.id === id);
+          return part
+            ? { ...acc, [id]: { x: part.position.x, y: part.position.y } }
+            : acc;
+        },
+        {} as Record<number, { x: number; y: number }>
+      );
+      originalPositionsRef.current = newOriginalPositions;
+    },
+    [gameBoardMode, selectedPartIds, selectMultipleParts, parts]
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)